arch/sparc/core/cpu_idle.c

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2018 ispace, inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+
+void k_cpu_idle(void)
+{
+	/* idle calls lock_irq() function before falling on this function.
+	 * but the unlock key is lost. so unlock irq unconditionally
+	 * irq_unlock function will enable unlock interrupt if key has
+	 * Enable Trap bit on.*/
+	irq_unlock(0);
```

Comment:
this function will immediately return - does SPARC have a "wait for interrupt" instruction because that's what this function needs to do

Reply:
Unfortunately it doesn't have the instruction.  Instead, it's a implementation dependent.  Some LEON3 has it in Ancillary State Register:

> 2.2.13 Power-down
> The processor can be configured to include a power-down feature to minimize power consumption during idle periods. The power-down mode is entered by performing a WRASR instruction to %asr19:
> ```
> wr %g0, %asr19
> ```
> During power-down, the pipeline is halted until the next interrupt occurs. Signals inside the processor pipeline and caches are then static, reducing power consumption from dynamic switching.

But this PR is for LEON **2**, which doesn't have it, neither.

Reply:
OK

Reply:
can you declare this __weak so it may be overridden in board or SOC configuration?

Reply:
Added __weak to `arch_cpu_idle()` and comment about it in the commit log.